test/index.ts

```diff
@@ -41,11 +41,11 @@ test('personalSign and recover', function (t) {
   const privKeyHex = '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0'
   const privKey = Buffer.from(privKeyHex, 'hex')
   const message = 'Hello, world!'
-  const msgParams = { data: message }
+  const msgParams: sigUtil.MsgParams<never> = { data: message }
```

Comment:
Since `TypedData<T>` is a union type of `string | EIP712TypedData | EIP712TypedData[] | TypedMessage<T>`, the `never` type here signifies that we know for sure that the `data` field will never be of the `TypedMessage<T>` type, so it's safe to use `never`.

I don't particularly like this syntax either, so I'm open to suggestion on improving this.

Reply:
> so it's safe to use `never`

I don't contest that it's _safe_ to use `never` here, I guess I'm more wondering if we can avoid using it in favour of something more semantic? I'll spend a bit more time thinking of an alternative.

Reply:
I have an idea -- instead of trying to parameterize on the `TypedData` type, we can instead parameterize on the API functions themselves, i.e.  make `TypedData` parameter-less, and then on the functions, require `MsgParams<TypedData | TypedMessage<T>>`.